bundles/com.salesforce.bazel.eclipse.common/src/main/java/com/salesforce/bazel/eclipse/logging/EclipseLoggerFacade.java

```diff
@@ -60,15 +59,67 @@
  * Unix platforms) if /tmp exists. It sends the log messages both to the official logger but also to this file which we
  * fully control.
  */
-public class EclipseLoggerFacade extends BasicLoggerFacade {
+public class EclipseLoggerFacade extends LoggerFacade {
     private static final Bundle BUNDLE = FrameworkUtil.getBundle(EclipseLoggerFacade.class);
     private static final ILog ECLIPSELOGGER = Platform.getLog(BUNDLE);
 
-    // we have had issues with hard to find logs, missing log lines, etc
-    // for Unix platorms, we create a custom log file in /tmp/bef.log to make sure we get our lines
+    // a custom log file in <user_home>/bef.log to duplicate logs in case of loss
     private static BufferedWriter befLogWriter = null;
     private static SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy-HH:mm:ss");
 
+    private static boolean extendedLog = true;
+
+    public static enum LogLevel {
+        DEBUG(IStatus.INFO, LoggerFacade.DEBUG, "DEBUG"),
+        INFO(IStatus.INFO, LoggerFacade.INFO, "INFO"),
+        ERROR(IStatus.ERROR, LoggerFacade.DEBUG, "ERROR"),
+        WARN(IStatus.WARNING, LoggerFacade.WARN, "WARN");
+
+        private final int severityCode;
+        private final int level;
+        private final String name;
+
+        private LogLevel(int severityCode, int level, String messagePrefix) {
+            this.severityCode = severityCode;
+            this.level = level;
+            this.name = messagePrefix;
+        }
+
+        public int getSeverityCode() {
+            return severityCode;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public int getLevel() {
+            return level;
+        }
+
+        public static LogLevel fromName(String name) {
+            for (LogLevel level : LogLevel.values()) {
+                if (level.getName().equalsIgnoreCase(name)) {
+                    return level;
+                }
+            }
+            return INFO; //default level
+        }
+    }
+
+    public static void setLevel(String name) {
+        LogLevel logLevel = LogLevel.fromName(name);
+        setLevel(logLevel.getLevel());
+    }
+
+    public static boolean isExtendedLog() {
```

Comment:
Why is `extended` needed? That's the difference between `debug` level and `info`+`extended`? Seems to complicated to me.

Reply:
initially, in SDK/BEF under Unix platform there were two logs: a standard logging and a custom logging in "bef.log" file and there was no any mechanism to switch this additional logging off. It resulted in disk space consumption, especially for debug level. The "extended" flag was introduced to have a possibility to control on/off the writing into "bef.log" file.